client.go

```diff
@@ -265,37 +253,192 @@ func (c *Client) CreateInstance(Name string, CPUs int, Memory int,
 	return instance, nil
 }
 
+// snapshotRequest defines options when making a snapshot of an instance
+type snapshotRequest struct {
+	All bool `json:"all"`
+}
+
+// SnapshotInstance takes a snapshot of an instance
+func (c Client) SnapshotInstance(instanceUUID string, all bool) error {
+	path := fmt.Sprintf("instances/%s/snapshot", instanceUUID)
+
+	request := &snapshotRequest{
+		All: all,
+	}
+	post, err := json.Marshal(request)
+	if err != nil {
+		return err
+	}
+
+	_, err = c.httpRequest(path, "POST", *bytes.NewBuffer(post))
+
+	return err
+}
+
+// Snapshot defines a snapshot of an instance
+type Snapshot struct {
+	UUID    string `json:"uuid"`
+	Device  string `json:"device"`
+	Created string `json:"created"` // TODO: Change to time.Time
```

Comment:
The result is a float Unix epoch time IIRC.

Reply:
Endpoint /nodes appears to return an RFC822 string in "lastseen".
Endpoint /instances/uuid/events returns a timestamp as a float?
My preference is for golang to convert the value to time.Time.

Reply:
Oh, nice catch on the inconsistency. So you're saying I should always just return epoch seconds?

Reply:
maybe milliseconds?

Reply:
ignore me. stick with seconds.

Reply:
Python likes epoch seconds as a float, so let's go with that. I've filed shakenfist/shakenfist#70 to track that.